client/my-sites/upgrades/trials/data/free-trial-cart.jsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import cartValues, { cartItems } from 'lib/cart-values';
+import observe from 'lib/mixins/data-observe';
+import TransactionStore from 'lib/transaction/store';
+import StoreConnection from 'components/data/store-connection';
+import { resetTransaction, setPayment } from 'lib/upgrades/actions';
+import storeTransactions from 'lib/store-transactions';
+
+const FreeTrialCartData = React.createClass( {
+	mixins: [ observe( 'sites', 'products' ) ],
+
+	getInitialState: function() {
+		return { cart: {} };
```

Comment:
It is a cart. It has the same structure, can be given to any `cart-values` function... It's just that it doesn't live in the `CartStore` so it really is temporary. The idea is to have this component handle the conversion from a `planName` (`premium` or `business`) to a valid cart object which will be given as props to its children and used in `checkout/transaction-steps-mixin.jsx` to make the actual checkout request. And since it also needs a valid `transaction` props this component also handles that.